### PR TITLE
Validate refunds against payment session in DB

### PR DESCRIPTION
**Pull request by rsachdevch**
__Short description outlining key changes/additions__
Use OrderCode in each refund to GetPaymentResourceByExternalID and validate each refund against a paymentSession in the DB.

__JIRA Ticket Number__
BI-10155

### Type of change

* [ ] Bug fix
* [x] New feature
* [ ] Breaking change

### Pull request checklist

* [x] I have added unit tests for new code that I have added
* [x] I have added/updated functional tests where appropriate
* [x] The code follows our [coding standards](https://github.com/companieshouse/styleguides/blob/master/go.md)

__An exhaustive list of peer review checks can be read [here](https://github.com/golang/go/wiki/CodeReviewComments)__

### CODE REVIEW

File: service/refund_test.go [hunk 1]
```diff
@@ -339,3 +340,155 @@ func TestUnitUpdateRefund(t *testing.T) {
...
+		batchRefund := generateXMLBatchRefund()
+
+		mockDao.EXPECT().GetPaymentResourceByExternalPaymentStatusID(gomock.Any()).Return(nil, fmt.Errorf("error")).AnyTimes()
+		validationErrors, err := service.ProcessGovPayBatchRefund(req.Context(), batchRefund)
+
+		So(len(validationErrors), ShouldEqual, 0)
+		So(err, ShouldNotBeNil)
+	})
+
+	Convey("Validation errors - payment session not founnd", t, func() {
```
Comment (Joseph-Nash) [inline service/refund_test.go:361]: should be "found"

File: service/refund.go [hunk 2]
```diff
@@ -128,3 +130,53 @@ func getRefundIndex(refunds []models.RefundResourceRest, refundId string) (int,
 	}
 	return -1, errors.New("refund id not found in payment refunds")
 }
+
+// ProcessGovPayBatchRefund retrieves all the payments in the batch refund
+// and validates it before processing it
+func (service *RefundService) ProcessGovPayBatchRefund(ctx context.Context, batchRefund models.GovPayRefundBatch) ([]string, error) {
+	var validationErrors []string
+	var mu = sync.Mutex{}
+	errs, ctx := errgroup.WithContext(ctx)
```
Comment (emead) [inline service/refund.go:139]: Rename `ctx` to new variable name to avoid conflicts

File: dao/mongo.go [hunk 3]
```diff
@@ -136,3 +137,28 @@ func (m *MongoService) PatchPaymentResource(id string, paymentUpdate *models.Pay
 
 	return err
 }
+
+// GetPaymentResourceByExternalPaymentStatusID retrieves a payment resource
+// associated with the externalPaymentStatusID provided
+func (m *MongoService) GetPaymentResourceByExternalPaymentStatusID(externalPaymentStatusID string) (*models.PaymentResourceDB, error) {
```
Comment (emead) [inline dao/mongo.go:143]: Need unit test for this function

File: handlers/bulk_refunds.go [hunk 4]
```diff
@@ -58,6 +60,22 @@ func HandleGovPayBulkRefund(w http.ResponseWriter, req *http.Request) {
 		return
 	}
 
+	// Validate batch refund request data against data in DB
+	validationErrors, err := refundService.ProcessGovPayBatchRefund(req.Context(), batchRefund)
+	if err != nil {
+		log.ErrorR(req, err)
```
Comment (emead) [inline handlers/bulk_refunds.go:66]: This error scenario is not covered by the unit tests

**Review by emead — APPROVED** [review]

**Pull request merged**